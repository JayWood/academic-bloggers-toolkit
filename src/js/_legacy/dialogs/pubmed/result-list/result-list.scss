@import 'colors';

.main {
    max-height: calc(100vh - 240px);
    overflow-y: auto;
    overflow-x: hidden;
}

.result {
    border-bottom: 1px solid $border;
}

.source {
    font-style: italic;
}

.button-group {
    display: flex;
}

.row-1 {
    padding: 0 16px;
    padding-top: 16px;
    font-size: 14px;
    line-height: 18px;
    display: flex;
    justify-content: space-between;
    color: $font-light;
}

.row-2 {
    font-size: 16px;
    line-height: 20px;
    padding: 8px 16px;
}

.row-3 {
    padding: 0 8px 4px;
    color: $font-light;
    line-height: 18px;
    font-size: 14px;
    padding-right: 8px;
    padding-left: 16px;
    display: flex;
    justify-content: space-between;
    align-items: center;
}
