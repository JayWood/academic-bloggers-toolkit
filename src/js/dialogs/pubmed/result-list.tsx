import { decode } from 'he';
import { observer } from 'mobx-react';
import * as React from 'react';

import { colors } from 'utils/styles';

import Button from 'components/button';

interface ResultListProps {
    /** List of results returned from pubmed search */
    results: PubMed.Response[];
    /** Callback to be performed when a result is chosen */
    onSelect(pmid: string): void;
}

@observer
export class ResultList extends React.PureComponent<ResultListProps, {}> {
    static readonly labels = top.ABT_i18n.dialogs.pubmed;

    /** Required so that result list can be scrolled to top after each new search */
    element: HTMLElement;

    bindRefs = (c: HTMLDivElement) => {
        this.element = c;
    };

    handleClick = (e: React.MouseEvent<HTMLInputElement>) => {
        this.props.onSelect(e.currentTarget.id);
    };

    componentDidUpdate() {
        this.element.scrollTop = 0;
    }

    render() {
        return (
            <div className="result-list" ref={this.bindRefs}>
                {this.props.results.map(result =>
                    <div key={result.uid} className="result-item">
                        <div className="result-item__row-1">
                            <div className="result-item__source" children={result.source} />
                            <div children={result.pubdate!.slice(0, 4)} />
                        </div>
                        {/* tslint:disable-next-line:react-no-dangerous-html */}
                        <div
                            dangerouslySetInnerHTML={{ __html: decode(result.title!) }}
                            className="result-item__row-2"
                        />
                        <div className="result-item__row-3">
                            <div
                                children={result.authors!.slice(0, 3).map(el => el.name).join(', ')}
                            />
                            <div className="result-item__buttons">
                                <Button
                                    flat
                                    focusable
                                    href={`https://www.ncbi.nlm.nih.gov/pubmed/${result.uid}`}
                                    label={ResultList.labels.viewReference}
                                />
                                <Button
                                    flat
                                    primary
                                    focusable
                                    id={result.uid}
                                    label={ResultList.labels.addReference}
                                    onClick={this.handleClick}
                                />
                            </div>
                        </div>
                    </div>,
                )}
                <style jsx>{`
                    .result-list {
                        max-height: calc(100vh - 240px);
                        overflow-y: auto;
                        overflow-x: hidden;
                    }
                    .result-item {
                        border-bottom: 1px solid ${colors.border};
                    }
                    .result-item__source {
                        font-style: italic;
                    }
                    .result-item__buttons {
                        display: flex;
                    }
                    .result-item__row-1 {
                        padding: 0 16px;
                        padding-top: 16px;
                        font-size: 14px;
                        line-height: 18px;
                        display: flex;
                        justify-content: space-between;
                        color: ${colors.font_light};
                    }
                    .result-item__row-2 {
                        padding: 0 16px;
                        font-size: 16px;
                        line-height: 20px;
                        padding: 8px 16px;
                    }
                    .result-item__row-3 {
                        padding: 0 8px 4px;
                        color: ${colors.font_light};
                        line-height: 18px;
                        font-size: 14px;
                        padding-right: 8px;
                        padding-left: 16px;
                        display: flex;
                        justify-content: space-between;
                    }
                `}</style>
            </div>
        );
    }
}
