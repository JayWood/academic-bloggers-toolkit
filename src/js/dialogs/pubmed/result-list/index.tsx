import { decode } from 'he';
import { observer } from 'mobx-react';
import * as React from 'react';

import Button from 'components/button';
import * as styles from './result-list.scss';

interface Props {
    /** List of results returned from pubmed search */
    results: CSL.Data[];
    /** Callback to be performed when a result is chosen */
    onSelect(pmid: string): void;
}

@observer
export default class ResultList extends React.Component<Props> {
    static readonly labels = top.ABT.i18n.dialogs.pubmed;

    /** Required so that result list can be scrolled to top after each new search */
    element: HTMLElement;

    bindRefs = (c: HTMLDivElement): void => {
        this.element = c;
    };

    componentDidUpdate(): void {
        this.element.scrollTop = 0;
    }

    handleClick = (e: React.MouseEvent<HTMLInputElement>): void => {
        this.props.onSelect(e.currentTarget.id);
    };

    handleWheel = (e: React.WheelEvent<HTMLDivElement>): void => {
        const isScrollingDown = e.deltaY > 0;
        const isScrollingUp = !isScrollingDown;
        const isScrollable =
            e.currentTarget.scrollHeight > e.currentTarget.clientHeight;
        const atBottom =
            e.currentTarget.scrollHeight <=
            e.currentTarget.clientHeight + Math.ceil(e.currentTarget.scrollTop);
        const atTop = e.currentTarget.scrollTop === 0;

        if (isScrollable && !atBottom && isScrollingDown) {
            e.cancelable = false;
        }

        if (isScrollable && !atTop && isScrollingUp) {
            e.cancelable = false;
        }
    };

    render(): JSX.Element {
        return (
            <div
                className={styles.main}
                ref={this.bindRefs}
                onWheel={this.handleWheel}
            >
                {this.props.results.map(result => (
                    <div key={result.PMID} className={styles.result}>
                        <div className={styles.row1}>
                            <span className={styles.source}>
                                {result.journalAbbreviation}
                            </span>
                            <span>{result.issued!['date-parts']![0][0]}</span>
                        </div>
                        {/* tslint:disable-next-line:react-no-dangerous-html */}
                        <div
                            dangerouslySetInnerHTML={{
                                __html: decode(result.title!),
                            }}
                            className={styles.row2}
                        />
                        <div className={styles.row3}>
                            <div>
                                {/* FIXME: This should probably be extracted to a component/function */}
                                {result.author!
                                    .slice(0, 3)
                                    .map(
                                        el =>
                                            `${el.family}, ${el.given!
                                                .charAt(0)
                                                .toUpperCase()}`,
                                    )
                                    .join(', ')}
                            </div>
                            <div className={styles.buttonGroup}>
                                <Button
                                    flat
                                    focusable
                                    href={`https://www.ncbi.nlm.nih.gov/pubmed/${
                                        result.PMID
                                    }`}
                                    label={ResultList.labels.view_reference}
                                />
                                <Button
                                    flat
                                    primary
                                    focusable
                                    id={result.PMID}
                                    label={ResultList.labels.add_reference}
                                    onClick={this.handleClick}
                                />
                            </div>
                        </div>
                    </div>
                ))}
            </div>
        );
    }
}
