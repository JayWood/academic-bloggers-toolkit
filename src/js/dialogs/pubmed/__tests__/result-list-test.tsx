import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as React from 'react';
import { ResultList } from '../result-list';

const generateData = (n: number) => {
    let data: Array<Partial<CSL.Data>> = [];
    for (let i = 0; i < n; i++) {
        data = [
            ...data,
            {
                author: [
                    { family: 'Doe', given: 'John' },
                    { family: 'Smith', given: 'Jan' },
                    { family: 'Brown', given: 'Pamela' },
                    { family: 'Clark', given: 'Steve' },
                ],
                issued: {
                    'date-parts': [['2016', '05', '01']],
                },
                'container-title-short': `J Test ${i + 1}`,
                title: `Test Title ${i + 1}`,
                id: `${i + 1}`,
            },
        ];
    }
    return data;
};

const defaultData = generateData(2);

const setup = (data: any = defaultData, fullMount = true) => {
    const spy = jest.fn();
    const component = fullMount
        ? mount(<ResultList results={data} onSelect={spy} />)
        : shallow(<ResultList results={data} onSelect={spy} />);
    return {
        spy,
        component,
        instance: component.instance() as any,
        results: component.find('.result-item'),
    };
};

describe('<ResultList />', () => {
    it('should render with two children', () => {
        const { results } = setup();
        expect(results.length).toBe(2);
    });
    it('should call handleClick on addReference click', () => {
        const { results, spy } = setup();
        expect(spy).toHaveBeenCalledTimes(0);
        const button = results
            .children()
            .find('Button')
            .at(1)
            .find('button');
        button.simulate('click');
        expect(spy).toHaveBeenCalledTimes(1);
        expect(spy.mock.calls[0]).toEqual(['1']);
    });
    it('should scroll to the top on update', () => {
        const { instance } = setup();
        instance.element.scrollTop = 150;
        instance.componentDidUpdate();
        expect(instance.element.scrollTop).toBe(0);
    });
    it('should match snapshots', () => {
        let { component } = setup();
        expect(toJSON(component)).toMatchSnapshot();

        ({ component } = setup([]));
        expect(toJSON(component)).toMatchSnapshot();
    });
    it('should show only 5 results at a time', () => {
        const { component } = setup(generateData(50));
        expect(component).toBeTruthy();
    });
    describe('wheel event tests', () => {
        const { component } = setup(undefined, false);
        const wheelDiv = component.find('.result-list');
        let atTop: any = {};
        let inMiddle: any = {};
        let atBottom: any = {};
        beforeEach(() => {
            atTop = {
                cancelable: true,
                deltaY: 0,
                currentTarget: {
                    clientHeight: 50,
                    scrollHeight: 100,
                    scrollTop: 0,
                },
            };
            inMiddle = {
                cancelable: true,
                deltaY: 0,
                currentTarget: {
                    clientHeight: 50,
                    scrollHeight: 100,
                    scrollTop: 25,
                },
            };
            atBottom = {
                cancelable: true,
                deltaY: 0,
                currentTarget: {
                    clientHeight: 50,
                    scrollHeight: 100,
                    scrollTop: 50,
                },
            };
        });
        it('should cancel at top scrolling up', () => {
            atTop.deltaY = -5;
            wheelDiv.simulate('wheel', atTop);
            expect(atTop.cancelable).toBe(true);
        });
        it('should cancel at bottom scrolling down', () => {
            atBottom.deltaY = 5;
            wheelDiv.simulate('wheel', atBottom);
            expect(atBottom.cancelable).toBe(true);
        });
        it('should NOT cancel in middle scrolling up', () => {
            inMiddle.deltaY = -5;
            wheelDiv.simulate('wheel', inMiddle);
            expect(inMiddle.cancelable).toBe(false);
        });
        it('should NOT cancel in middle scrolling down', () => {
            inMiddle.deltaY = 5;
            wheelDiv.simulate('wheel', inMiddle);
            expect(inMiddle.cancelable).toBe(false);
        });
    });
});
