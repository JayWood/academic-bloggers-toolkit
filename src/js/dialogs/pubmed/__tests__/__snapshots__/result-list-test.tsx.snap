// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultList /> should match snapshots 1`] = `
<ResultList
  onSelect={[Function]}
  results={
    Array [
      Object {
        "authors": Array [
          Object {
            "authtype": "Author",
            "name": "Author 1",
          },
          Object {
            "authtype": "Author",
            "name": "Author 2",
          },
          Object {
            "authtype": "Author",
            "name": "Author 3",
          },
          Object {
            "authtype": "Author",
            "name": "NOT VISIBLE",
          },
        ],
        "pubdate": "2016 May 1",
        "source": "J Test 1",
        "title": "Test Title 1",
        "uid": "1",
      },
      Object {
        "authors": Array [
          Object {
            "authtype": "Author",
            "name": "Author 1",
          },
          Object {
            "authtype": "Author",
            "name": "Author 2",
          },
          Object {
            "authtype": "Author",
            "name": "Author 3",
          },
          Object {
            "authtype": "Author",
            "name": "NOT VISIBLE",
          },
        ],
        "pubdate": "2016 May 1",
        "source": "J Test 2",
        "title": "Test Title 2",
        "uid": "2",
      },
    ]
  }
>
  <div
    className="result-list"
    onWheel={[Function]}
  >
    <div
      className="result-item"
    >
      <div
        className="result-item__row-1"
      >
        <div
          className="result-item__source"
        >
          J Test 1
        </div>
        <div>
          2016
        </div>
      </div>
      <div
        className="result-item__row-2"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Test Title 1",
          }
        }
      />
      <div
        className="result-item__row-3"
      >
        <div>
          Author 1, Author 2, Author 3
        </div>
        <div
          className="result-item__buttons"
        >
          <Button
            flat={true}
            focusable={true}
            href="https://www.ncbi.nlm.nih.gov/pubmed/1"
            label="View"
            type="button"
          >
            <div>
              <button
                className="focusable btn-flat"
                onClick={[Function]}
                type="button"
              >
                View
              </button>
              <style
                jsx={true}
              >
                
                    button {
                        -webkit-appearance: none;
                        -webkit-touch-callout: none;
                        -webkit-user-select: none;
                        -khtml-user-select: none;
                        -moz-user-select: none;
                        -ms-user-select: none;
                        user-select: none;
                        background: white;
                        box-sizing: border-box;
                        color: #555555;
                        background-image: none;
                        border: none;
                        border-radius: 2px;
                        box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24);
                        cursor: pointer;
                        display: inline-block;
                        font-size: 13px;
                        font-weight: 500;
                        height: 36px;
                        line-height: 36px;
                        min-width: 64px;
                        outline: 0;
                        padding: 0 16px;
                        text-align: center;
                        text-decoration: none;
                        text-transform: uppercase;
                        transition: 
            box-shadow .2s cubic-bezier(0.4, 0, 1, 1),
            background-color .2s cubic-bezier(0.4, 0, 0.2, 1),
            color .2s cubic-bezier(0.4, 0, 0.2, 1)
        ;
                        vertical-align: middle;
                        white-space: nowrap;
                    }
                    button:hover {
                        box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                    }
                    button:active {
                        box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                    }
                    button:active:hover {
                        box-shadow: 0 10px 20px rgba(0, 0, 0, .19), 0 6px 6px rgba(0, 0, 0, .23);
                        background: rgba(0, 0, 0, 0.12);
                    }
                    button:disabled {
                        pointer-events: none !important;
                        color: #a0a5aa !important;
                    }
                    button.focusable:focus {
                        outline: 1.1px auto rgba(19, 124, 189, 0.5);
                        outline-offset: 2px;
                    }
                    button.focusable:focus:active {
                        outline: 0;
                    }
                    .btn-flat,
                    .btn-flat.btn-primary {
                        background: rgba(0, 0, 0, 0);
                        box-shadow: none !important;
                    }
                    .btn-flat:hover,
                    .btn-flat.btn-primary:hover {
                        background: rgba(158, 158, 158, 0.2);
                    }
                    .btn-flat:active,
                    .btn-flat.btn-primary:active {
                        background: rgba(13, 13, 13, 0.12);
                    }
                    .btn-flat:hover:active,
                    .btn-flat.btn-primary:hover:active {
                        background: rgba(0, 0, 0, 0.12);
                    }
                    .btn-flat.btn-primary {
                        color: #0085ba;
                    }
                    .btn-primary {
                        background: #0085ba;
                        color: white;
                    }
                    .btn-primary:active:hover {
                        background: #007aab;
                    }
                    .btn-primary:disabled {
                        opacity: 0.6;
                        box-shadow: none;
                        color: white !important;
                    }
                    .btn-icon {
                        border-radius: 50%;
                        font-size: 24px;
                        height: 32px;
                        margin: auto 0;
                        min-width: 32px;
                        width: 32px;
                        padding: 0;
                        line-height: normal;
                    }
                    .btn-icon:hover {
                        color: #555555;
                    }

                    .dashicons {
                        line-height: 32px !important;
                        width: 32px;
                        height: 32px;
                    }
                    .dashicons-migrate {
                        transform: rotateY(180deg);
                        padding-left: 5px;
                    }
                    .dashicons-minus {
                        font-weight: 900;
                    }
                    .dashicons-menu,
                    .dashicons-no-alt {
                        font-size: 25px;
                    }
                    .dashicons-plus {
                        padding-top: 2px;
                    }
                
              </style>
            </div>
          </Button>
          <Button
            flat={true}
            focusable={true}
            id="1"
            label="Select"
            onClick={[Function]}
            primary={true}
            type="button"
          >
            <div>
              <button
                className="focusable btn-primary btn-flat"
                id="1"
                onClick={[Function]}
                type="button"
              >
                Select
              </button>
              <style
                jsx={true}
              >
                
                    button {
                        -webkit-appearance: none;
                        -webkit-touch-callout: none;
                        -webkit-user-select: none;
                        -khtml-user-select: none;
                        -moz-user-select: none;
                        -ms-user-select: none;
                        user-select: none;
                        background: white;
                        box-sizing: border-box;
                        color: #555555;
                        background-image: none;
                        border: none;
                        border-radius: 2px;
                        box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24);
                        cursor: pointer;
                        display: inline-block;
                        font-size: 13px;
                        font-weight: 500;
                        height: 36px;
                        line-height: 36px;
                        min-width: 64px;
                        outline: 0;
                        padding: 0 16px;
                        text-align: center;
                        text-decoration: none;
                        text-transform: uppercase;
                        transition: 
            box-shadow .2s cubic-bezier(0.4, 0, 1, 1),
            background-color .2s cubic-bezier(0.4, 0, 0.2, 1),
            color .2s cubic-bezier(0.4, 0, 0.2, 1)
        ;
                        vertical-align: middle;
                        white-space: nowrap;
                    }
                    button:hover {
                        box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                    }
                    button:active {
                        box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                    }
                    button:active:hover {
                        box-shadow: 0 10px 20px rgba(0, 0, 0, .19), 0 6px 6px rgba(0, 0, 0, .23);
                        background: rgba(0, 0, 0, 0.12);
                    }
                    button:disabled {
                        pointer-events: none !important;
                        color: #a0a5aa !important;
                    }
                    button.focusable:focus {
                        outline: 1.1px auto rgba(19, 124, 189, 0.5);
                        outline-offset: 2px;
                    }
                    button.focusable:focus:active {
                        outline: 0;
                    }
                    .btn-flat,
                    .btn-flat.btn-primary {
                        background: rgba(0, 0, 0, 0);
                        box-shadow: none !important;
                    }
                    .btn-flat:hover,
                    .btn-flat.btn-primary:hover {
                        background: rgba(158, 158, 158, 0.2);
                    }
                    .btn-flat:active,
                    .btn-flat.btn-primary:active {
                        background: rgba(13, 13, 13, 0.12);
                    }
                    .btn-flat:hover:active,
                    .btn-flat.btn-primary:hover:active {
                        background: rgba(0, 0, 0, 0.12);
                    }
                    .btn-flat.btn-primary {
                        color: #0085ba;
                    }
                    .btn-primary {
                        background: #0085ba;
                        color: white;
                    }
                    .btn-primary:active:hover {
                        background: #007aab;
                    }
                    .btn-primary:disabled {
                        opacity: 0.6;
                        box-shadow: none;
                        color: white !important;
                    }
                    .btn-icon {
                        border-radius: 50%;
                        font-size: 24px;
                        height: 32px;
                        margin: auto 0;
                        min-width: 32px;
                        width: 32px;
                        padding: 0;
                        line-height: normal;
                    }
                    .btn-icon:hover {
                        color: #555555;
                    }

                    .dashicons {
                        line-height: 32px !important;
                        width: 32px;
                        height: 32px;
                    }
                    .dashicons-migrate {
                        transform: rotateY(180deg);
                        padding-left: 5px;
                    }
                    .dashicons-minus {
                        font-weight: 900;
                    }
                    .dashicons-menu,
                    .dashicons-no-alt {
                        font-size: 25px;
                    }
                    .dashicons-plus {
                        padding-top: 2px;
                    }
                
              </style>
            </div>
          </Button>
        </div>
      </div>
    </div>
    <div
      className="result-item"
    >
      <div
        className="result-item__row-1"
      >
        <div
          className="result-item__source"
        >
          J Test 2
        </div>
        <div>
          2016
        </div>
      </div>
      <div
        className="result-item__row-2"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Test Title 2",
          }
        }
      />
      <div
        className="result-item__row-3"
      >
        <div>
          Author 1, Author 2, Author 3
        </div>
        <div
          className="result-item__buttons"
        >
          <Button
            flat={true}
            focusable={true}
            href="https://www.ncbi.nlm.nih.gov/pubmed/2"
            label="View"
            type="button"
          >
            <div>
              <button
                className="focusable btn-flat"
                onClick={[Function]}
                type="button"
              >
                View
              </button>
              <style
                jsx={true}
              >
                
                    button {
                        -webkit-appearance: none;
                        -webkit-touch-callout: none;
                        -webkit-user-select: none;
                        -khtml-user-select: none;
                        -moz-user-select: none;
                        -ms-user-select: none;
                        user-select: none;
                        background: white;
                        box-sizing: border-box;
                        color: #555555;
                        background-image: none;
                        border: none;
                        border-radius: 2px;
                        box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24);
                        cursor: pointer;
                        display: inline-block;
                        font-size: 13px;
                        font-weight: 500;
                        height: 36px;
                        line-height: 36px;
                        min-width: 64px;
                        outline: 0;
                        padding: 0 16px;
                        text-align: center;
                        text-decoration: none;
                        text-transform: uppercase;
                        transition: 
            box-shadow .2s cubic-bezier(0.4, 0, 1, 1),
            background-color .2s cubic-bezier(0.4, 0, 0.2, 1),
            color .2s cubic-bezier(0.4, 0, 0.2, 1)
        ;
                        vertical-align: middle;
                        white-space: nowrap;
                    }
                    button:hover {
                        box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                    }
                    button:active {
                        box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                    }
                    button:active:hover {
                        box-shadow: 0 10px 20px rgba(0, 0, 0, .19), 0 6px 6px rgba(0, 0, 0, .23);
                        background: rgba(0, 0, 0, 0.12);
                    }
                    button:disabled {
                        pointer-events: none !important;
                        color: #a0a5aa !important;
                    }
                    button.focusable:focus {
                        outline: 1.1px auto rgba(19, 124, 189, 0.5);
                        outline-offset: 2px;
                    }
                    button.focusable:focus:active {
                        outline: 0;
                    }
                    .btn-flat,
                    .btn-flat.btn-primary {
                        background: rgba(0, 0, 0, 0);
                        box-shadow: none !important;
                    }
                    .btn-flat:hover,
                    .btn-flat.btn-primary:hover {
                        background: rgba(158, 158, 158, 0.2);
                    }
                    .btn-flat:active,
                    .btn-flat.btn-primary:active {
                        background: rgba(13, 13, 13, 0.12);
                    }
                    .btn-flat:hover:active,
                    .btn-flat.btn-primary:hover:active {
                        background: rgba(0, 0, 0, 0.12);
                    }
                    .btn-flat.btn-primary {
                        color: #0085ba;
                    }
                    .btn-primary {
                        background: #0085ba;
                        color: white;
                    }
                    .btn-primary:active:hover {
                        background: #007aab;
                    }
                    .btn-primary:disabled {
                        opacity: 0.6;
                        box-shadow: none;
                        color: white !important;
                    }
                    .btn-icon {
                        border-radius: 50%;
                        font-size: 24px;
                        height: 32px;
                        margin: auto 0;
                        min-width: 32px;
                        width: 32px;
                        padding: 0;
                        line-height: normal;
                    }
                    .btn-icon:hover {
                        color: #555555;
                    }

                    .dashicons {
                        line-height: 32px !important;
                        width: 32px;
                        height: 32px;
                    }
                    .dashicons-migrate {
                        transform: rotateY(180deg);
                        padding-left: 5px;
                    }
                    .dashicons-minus {
                        font-weight: 900;
                    }
                    .dashicons-menu,
                    .dashicons-no-alt {
                        font-size: 25px;
                    }
                    .dashicons-plus {
                        padding-top: 2px;
                    }
                
              </style>
            </div>
          </Button>
          <Button
            flat={true}
            focusable={true}
            id="2"
            label="Select"
            onClick={[Function]}
            primary={true}
            type="button"
          >
            <div>
              <button
                className="focusable btn-primary btn-flat"
                id="2"
                onClick={[Function]}
                type="button"
              >
                Select
              </button>
              <style
                jsx={true}
              >
                
                    button {
                        -webkit-appearance: none;
                        -webkit-touch-callout: none;
                        -webkit-user-select: none;
                        -khtml-user-select: none;
                        -moz-user-select: none;
                        -ms-user-select: none;
                        user-select: none;
                        background: white;
                        box-sizing: border-box;
                        color: #555555;
                        background-image: none;
                        border: none;
                        border-radius: 2px;
                        box-shadow: 0 1px 3px rgba(0, 0, 0, .12), 0 1px 2px rgba(0, 0, 0, .24);
                        cursor: pointer;
                        display: inline-block;
                        font-size: 13px;
                        font-weight: 500;
                        height: 36px;
                        line-height: 36px;
                        min-width: 64px;
                        outline: 0;
                        padding: 0 16px;
                        text-align: center;
                        text-decoration: none;
                        text-transform: uppercase;
                        transition: 
            box-shadow .2s cubic-bezier(0.4, 0, 1, 1),
            background-color .2s cubic-bezier(0.4, 0, 0.2, 1),
            color .2s cubic-bezier(0.4, 0, 0.2, 1)
        ;
                        vertical-align: middle;
                        white-space: nowrap;
                    }
                    button:hover {
                        box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                    }
                    button:active {
                        box-shadow: 0 3px 6px rgba(0, 0, 0, .16), 0 3px 6px rgba(0, 0, 0, .23);
                    }
                    button:active:hover {
                        box-shadow: 0 10px 20px rgba(0, 0, 0, .19), 0 6px 6px rgba(0, 0, 0, .23);
                        background: rgba(0, 0, 0, 0.12);
                    }
                    button:disabled {
                        pointer-events: none !important;
                        color: #a0a5aa !important;
                    }
                    button.focusable:focus {
                        outline: 1.1px auto rgba(19, 124, 189, 0.5);
                        outline-offset: 2px;
                    }
                    button.focusable:focus:active {
                        outline: 0;
                    }
                    .btn-flat,
                    .btn-flat.btn-primary {
                        background: rgba(0, 0, 0, 0);
                        box-shadow: none !important;
                    }
                    .btn-flat:hover,
                    .btn-flat.btn-primary:hover {
                        background: rgba(158, 158, 158, 0.2);
                    }
                    .btn-flat:active,
                    .btn-flat.btn-primary:active {
                        background: rgba(13, 13, 13, 0.12);
                    }
                    .btn-flat:hover:active,
                    .btn-flat.btn-primary:hover:active {
                        background: rgba(0, 0, 0, 0.12);
                    }
                    .btn-flat.btn-primary {
                        color: #0085ba;
                    }
                    .btn-primary {
                        background: #0085ba;
                        color: white;
                    }
                    .btn-primary:active:hover {
                        background: #007aab;
                    }
                    .btn-primary:disabled {
                        opacity: 0.6;
                        box-shadow: none;
                        color: white !important;
                    }
                    .btn-icon {
                        border-radius: 50%;
                        font-size: 24px;
                        height: 32px;
                        margin: auto 0;
                        min-width: 32px;
                        width: 32px;
                        padding: 0;
                        line-height: normal;
                    }
                    .btn-icon:hover {
                        color: #555555;
                    }

                    .dashicons {
                        line-height: 32px !important;
                        width: 32px;
                        height: 32px;
                    }
                    .dashicons-migrate {
                        transform: rotateY(180deg);
                        padding-left: 5px;
                    }
                    .dashicons-minus {
                        font-weight: 900;
                    }
                    .dashicons-menu,
                    .dashicons-no-alt {
                        font-size: 25px;
                    }
                    .dashicons-plus {
                        padding-top: 2px;
                    }
                
              </style>
            </div>
          </Button>
        </div>
      </div>
    </div>
    <style
      jsx={true}
    >
      
                    .result-list {
                        max-height: calc(100vh - 240px);
                        overflow-y: auto;
                        overflow-x: hidden;
                    }
                    .result-item {
                        border-bottom: 1px solid rgba(0, 0, 0, 0.12);
                    }
                    .result-item__source {
                        font-style: italic;
                    }
                    .result-item__buttons {
                        display: flex;
                    }
                    .result-item__row-1 {
                        padding: 0 16px;
                        padding-top: 16px;
                        font-size: 14px;
                        line-height: 18px;
                        display: flex;
                        justify-content: space-between;
                        color: rgba(0, 0, 0, 0.54);
                    }
                    .result-item__row-2 {
                        padding: 0 16px;
                        font-size: 16px;
                        line-height: 20px;
                        padding: 8px 16px;
                    }
                    .result-item__row-3 {
                        padding: 0 8px 4px;
                        color: rgba(0, 0, 0, 0.54);
                        line-height: 18px;
                        font-size: 14px;
                        padding-right: 8px;
                        padding-left: 16px;
                        display: flex;
                        justify-content: space-between;
                    }
                
    </style>
  </div>
</ResultList>
`;

exports[`<ResultList /> should match snapshots 2`] = `
<ResultList
  onSelect={[Function]}
  results={Array []}
>
  <div
    className="result-list"
    onWheel={[Function]}
  >
    <style
      jsx={true}
    >
      
                    .result-list {
                        max-height: calc(100vh - 240px);
                        overflow-y: auto;
                        overflow-x: hidden;
                    }
                    .result-item {
                        border-bottom: 1px solid rgba(0, 0, 0, 0.12);
                    }
                    .result-item__source {
                        font-style: italic;
                    }
                    .result-item__buttons {
                        display: flex;
                    }
                    .result-item__row-1 {
                        padding: 0 16px;
                        padding-top: 16px;
                        font-size: 14px;
                        line-height: 18px;
                        display: flex;
                        justify-content: space-between;
                        color: rgba(0, 0, 0, 0.54);
                    }
                    .result-item__row-2 {
                        padding: 0 16px;
                        font-size: 16px;
                        line-height: 20px;
                        padding: 8px 16px;
                    }
                    .result-item__row-3 {
                        padding: 0 8px 4px;
                        color: rgba(0, 0, 0, 0.54);
                        line-height: 18px;
                        font-size: 14px;
                        padding-right: 8px;
                        padding-left: 16px;
                        display: flex;
                        justify-content: space-between;
                    }
                
    </style>
  </div>
</ResultList>
`;
